### ISSUE TITLE
Consider increasing granularity of the leaves by splitting days into hours.

### ISSUE THREAD

**Issue by vpp** [timeoff-management/application#37]
As one of our customers suggested:

"> I can provide one suggestion for a feature. It’d probably require a
> bit of an overhaul and rethinking of some things, but it seems
> important if you want to continue growing this for wider usage. Most
> places I’ve worked for have broken down personal and vacation time
> to hours, not days. Right now it seems as if your system entirely
> considers things in the concept of day, and not hours, and that would
> need to be reworked before we can deploy it (we have some accruals
> like 4.75 hours, not in even “day” amounts). Do you think this is
> something you’d ever consider moving to, or are you fairly stuck on
> the unit being days? "

We need to consider that.

**Comment by scancreeder** [comment]
I believe this to be a really good change - I've found in my experience that most companies track vacation, personal time, etc in hours, not days. This allows someone to request "4.5 hours" off, and accrue, say, "4.25 hours of personal time, per week worked".

My suggestion would be to change the unit to hours, and allow increments as granular as 0.25 hours (15 minutes). On the requesting time off page, you would have them enter a date, and then a text box for hours off, or you could have them enter a date-time range, and calculate how many hours that is.

**Comment by vpp** [comment]
I thought about the feature of increasing granularity of absences from half a day to the level of hours.

Tried to understand what is a problem we need to solve. 

A) Is it to enable employees to book just a few hours of off time?

B) Or to be able to spend time that is accrual up to current date?

In majority of cases company allocate a particular number of days per year an employee could take as a vacation. And an employee is free to pickup any amount of days off at any point during the year until the total number of vacation days stay within allowance. So employer can kind of credit employee allowing to spend entire allowance let's say within January and February.

On other hand I assume there are organisations that even though they have an allocation of certain number of vacation days per year, they release available vacations as year passes. Pretty much like they with salary. For instance let's say allocation contains 24 days per year, if available balances are released on monthly basis it will be 2 days per month per year; if it is on weekly basis - roughly half a day per week.

Back to the TimeOff.Management.

Solving the issue A) sounds quite complex at this stage. As it would involve alternations to the database schema and amendments of logic that checks integrity of leave requests, as well as calculation of remaining and used balances.

On other hand if we need to solve issue outlined at B), it is a matter of adding logic that will make sure that company does not credit employees with more vacations that are due till current point. Which is purely business logic matter and could be easily turned on/off for different companies.

**Comment by kenshi1023** [comment]
can we pay or make a donation for this feature to be implemented?

**Comment by vpp** [comment]
Absolutely, just drop me email at pavlo@timeoff.management for further discussion.

**Comment by 53c70r** [comment]
What is the status and priority of this topic?

Is there a way to configure the exact time when day_part_morning and leave_day_part_afternoon starts so you would get the correct time when exporting to the calendar?
